clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -76,54 +88,78 @@ public short version() {
     /**
      * Factory method for getting a request object based on ApiKey ID and a buffer
      */
-    public static AbstractRequest getRequest(int requestId, short versionId, ByteBuffer buffer) {
+    public static RequestAndSize getRequest(int requestId, short version, ByteBuffer buffer) {
```

Comment:
Makes sense. We can always rename the class if we need to add some more info in there.